### ISSUE TITLE
salt 2015.8.10 incompatible with pip v8.1.2

### ISSUE THREAD

**Issue by seanjnkns** [saltstack/salt#34595]
### Description of Issue/Question
After OpenStack Mitaka (CentOS 7.x distro) released python-pip 8.1.2-1.el7, pip.installed no longer works and throws errors.  Assumed this was fixed in #33163, but appears to not be the case.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
```
python-senlinclient:
  pip.installed
```
### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)
```
From Mitaka's python-pip:
          ID: python-senlinclient
    Function: pip.installed
      Result: False
     Comment: Error installing 'python-senlinclient': Traceback (most recent call last):
                File "/bin/pip2", line 9, in <module>
                  load_entry_point('pip==8.1.2', 'console_scripts', 'pip2')()
                File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 542, in load_entry_point
                  return get_distribution(dist).load_entry_point(group, name)
                File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2568, in load_entry_point
                  raise ImportError("Entry point %r not found" % ((group, name),))
              ImportError: Entry point ('console_scripts', 'pip2') not found
     Started: 21:23:38.280876
    Duration: 232.161 ms
     Changes:

Install python-pip 8.1.2 from source and getting the following:
          ID: python-senlinclient
    Function: pip.installed
      Result: False
     Comment: Error installing 'python-senlinclient': Exception:
              Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/pip/basecommand.py", line 215, in main
                  status = self.run(options, args)
                File "/usr/lib/python2.7/site-packages/pip/commands/freeze.py", line 85, in run
                  for line in freeze(**freeze_kwargs):
                File "/usr/lib/python2.7/site-packages/pip/operations/freeze.py", line 47, in freeze
                  dependency_links
                File "/usr/lib/python2.7/site-packages/pip/__init__.py", line 264, in from_dist
                  req = dist.as_requirement()
                File "/usr/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2562, in as_requirement
                  return Requirement.parse(spec)
                File "/usr/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2833, in parse
                  req, = parse_requirements(s)
                File "/usr/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2781, in parse_requirements
                  yield Requirement(line)
                File "/usr/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2790, in __init__
                  raise RequirementParseError(str(e))
              RequirementParseError: Invalid requirement, parse error at "'-ipap11h'"
```
### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2015.8.10

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: 0.21.1
           Mako: 0.8.1
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 1.5.2
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.2.3
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 4.4.4.bs.ufd
         system: CentOS Linux 7.2.1511 Core
```

**Comment by rallytime** [comment]
@seanjnkns Are you able to `pip install` the package directly? In am wondering if there is something wrong with the pip package you have on that box and this is an upstream bug.

I don't have a Mitaka version to test on currently, however, I am unable to replicate this issue on other distributions (tested with CentOS 7, Ubuntu 14.04, and Arch Linux) using a simple state with the latest version of pip available (Versions `8.1.2`). Here's the output from my test on a CentOS 7 box:

```
root@rallytime:~# cat /srv/salt/pip-test.sls
awscli:
  pip.installed
root@rallytime:~# salt rally-cent7 cmd.run 'pip -V'
rally-cent7:
    pip 8.1.2 from /usr/lib/python2.7/site-packages (python 2.7)
root@rallytime:~# salt rally-cent7 state.sls pip-test
rally-cent7:
----------
          ID: awscli
    Function: pip.installed
      Result: True
     Comment: All packages were successfully installed
     Started: 20:37:32.745567
    Duration: 5602.366 ms
     Changes:
              ----------
              awscli==1.10.47:
                  Installed

Summary for rally-cent7
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
root@rallytime:~# salt rally-cent7 test.versions_report
rally-cent7:
    Salt Version:
               Salt: 2015.8.10

    Dependency Versions:
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
               Mako: Not Installed
             PyYAML: 3.11
              PyZMQ: 14.7.0
             Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
               RAET: Not Installed
            Tornado: 4.2.1
                ZMQ: 4.0.5
               cffi: Not Installed
           cherrypy: Not Installed
           dateutil: 2.5.3
              gitdb: Not Installed
          gitpython: Not Installed
              ioflo: Not Installed
            libgit2: Not Installed
            libnacl: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.4.7
       mysql-python: Not Installed
          pycparser: Not Installed
           pycrypto: 2.6.1
             pygit2: Not Installed
       python-gnupg: Not Installed
              smmap: Not Installed
            timelib: Not Installed

    System Versions:
               dist: centos 7.0.1406 Core
            machine: x86_64
            release: 4.5.5-x86_64-linode69
             system: CentOS Linux 7.0.1406 Core
```

**Comment by seanjnkns** [comment]
Yeah, if I pip install it manually, it works fine.  If I roll back to a 7.1.0 version, the one we have in our own internal repos, it works fine too.  If I install the version from Mitaka, v8.1.2, it fails to install/check if it's installed.  If I re-downgrade to 7.1.0 where it's working, and then `pip install --upgrade pip` and try, it again fails.

**Comment by seanjnkns** [comment]
Also, @rallytime, I see you're trying to install a different module.  What happens if you try the same state I'm trying?

**Comment by seanjnkns** [comment]
Actually, seems an issue that was resolved via #3764.  I removed ipa-python and ipa-client, and now it works fine.  I'll need to create a state to ensure those 2 packages are absent and then it should be fine.

**Comment by seanjnkns** [comment]
If you want to investigate this further, you're welcome, or we can go ahead and close it.

**Comment by rallytime** [comment]
@seanjnkns The first two distros I tried (Ubuntu and Arch) I used your original state. I switch the package in the state on CentOS 7 because I was running into some different, unrelated errors. I wanted to make sure it wasn't a salt/pip problem vs. just that particular `python-selinclient` package.

If you're ok with closing it, then let's do that. I think if it pops up again, we can definitely re-open.
